app/ui/src/app/settings/oauth-apps/oauth-app-form.component.ts

```diff
@@ -15,18 +15,27 @@ import {
   DynamicInputModel
 } from '@ng-dynamic-forms/core';
 
-// Default form config object for OAuth client settings
-const OAUTH_APP_FORM_CONFIG = {
-  clientId: {
-    displayName: 'Client ID',
-    type: 'string',
-    description: 'The OAuth client ID setting for the target application'
-  },
-  clientSecret: {
-    displayName: 'Client Secret',
-    type: 'password',
-    description: 'The OAuth client secret value for the target application'
-  }
+const uniqueNum = () => {
+  console.log('uniqueNum called');
+  return Math.random().toString(36).substr(2, 16);
+}
+
+const uniqueIdProp = () => 'clientId-' + uniqueNum();
+const uniqueSecretProp = () => 'clientSecret-' + uniqueNum();
+const idName = uniqueIdProp();
+const secretName = uniqueSecretProp();
```

Comment:
IdName and secretName are assigned just once, so there is no point on leveraging lambda functions (uniqueIdProp and uniqueSecretPop) to generate its unique key. You can assign these values directly or even execute and assign uniqueNum just once and then append its value through a E6 template string to these two (and any future) key on OAUTH_APP-FORM_CONFIG

Reply:
Good call on template strings being a better approach. My question, about these values being assigned just a single time, was kind of the main point though. It's problematic because we end up with duplicate ID's for the form fields.

We need a way to make these ID's unique for each app in the list, until then there will be a hiccup in the UI when a user expands multiple of these containers and selects one of the labels - focus will likely be placed into the wrong input.

Maybe that should be addressed as a separate issue? For now I just put it back like it was before.

Reply:
Also note for this particular set of forms this may eventually be driven by the backend -> https://github.com/syndesisio/syndesis/issues/240

Also it might be best to handle the actual element IDs from within the form component itself rather than modifying the input?  I think code using the form service shouldn't need to worry about making unique element IDs, it's more a concern for the component that renders the form to the page.

Reply:
I added a new property ns "namespace" to the forms service createFormModel. This is used when present to prefix the id of the form controls. I wasn't sure how to handle this directly within the component view template, but as a service param it becomes sort of configurable which is nice. What do you think?

Reply:
Updated, changes to the form service reverted and now we're only applying the unique id's at the DOM level. Looking much cleaner, let me know what you guys think!